### ISSUE TITLE
Tags disappearing on rebuild/edit

### ISSUE THREAD

**Issue by aaronstonedd** [cfenollosa/bashblog#70]
Tags disappear when running ./bb.sh rebuild. Also, when editing any post (not the full html file, just the post content), there are no tags.

To reproduce the issue:
1. Create a new post (with the title "My New Post", for example) with tags in them.
2. Publish the post.
3. Either type "./bb.sh edit my-new-post.html" to edit the post, or "./bb.sh rebuild" to rebuild the entire blog.

You'll see that there are no tags any more.

**Comment by aaronstonedd** [comment]
I think I found the culprit, but I don't know what the mistake is.

Running
````
awk '/<!-- text begin -->/, /<!-- text end -->/{if(!/<!-- text begin -->/ && !/<!-- text end -->/) print}' < first-blog-post-hello-internet.html
````
gives me 
````
<p>This is the first blogpost on hypertext.website. I'm quite excited, 
as this means I can now start using the web space allocated to me on 
hypertext.website properly.</p>
````
for the web page http://hypertext.website/~dheeraj/blog/first-blog-post-hello-internet.html generated using bashblog.

**Comment by aaronstonedd** [comment]
I think I found what the mistake is. When I did `view-source:http://hypertext.website/~dheeraj/blog/first-blog-post-hello-internet.html` on Chrome, I found the following:
````
<p>Tags: <a href='tag_maintenance.html'>maintenance</a>, <a href='tag_personal.html'>personal</a></p><!-- text end -->
````
This means that there is no newline between </p> and , so the whole line containing  is discarded. This affects both editing and rebuilding (I checked both functions edit() and the one for rebuilding, and both use `get_html_file_content "text" "text"`, reproducing the bug in question).

So the fix would be to make  come on a new line (like ).

**Comment by aaronstonedd** [comment]
If you look at the following
(
````
        echo "$global_author</div>" >> "$filename"
        echo '<!-- text begin -->' >> "$filename" # This marks the text body, after the title, date...
    fi
    cat "$content" >> "$filename" # Actual content
    if [[ "$index" == "no" ]]; then
        echo '<!-- text end -->' >> "$filename"
        twitter "$global_url/$file_url" >> "$filename"
        echo '<!-- entry end -->' >> "$filename" # absolute end of the post
    fi
    echo '</div>' >> "$filename" # content
````
), you will find that the problem arises because `cat "$content" >> "$filename" # Actual content` followed by `echo '<!-- text end-->' >> "$filename"` does not mean that a newline was inserted between the two statements.

So the best (foolproof) solution would be to write '\n\n' and '\n\n' (and the same for beginnings too, because the current proper behaviour depends on the previous statement being an echo statement too).

**Comment by aaronstonedd** [comment]
Please see the changes in the file below (it fixes this issue, and another issue that comes because of the EDITOR component not being used when rebuilding entries – inadvertently, an extra newline is inserted between `<p>Tags: ...</p>` and `<!-- text end -->`, which is also fixed).

The updated version of file bb.sh is at the following link: 
http://cl.ly/code/3o42252m162L

**Comment by cfenollosa** [comment]
Thanks for the bug report! 

Let me a few days to look at it. I'll try to push a fix.

Carlos

**Comment by cfenollosa** [comment]
Hello,

I can't reproduce the bug.

I've created a post, and then run `rebuild` and `edit` multiple times, and the tags are always there.

Could you tell me which OS are you using? It may be an `awk` incompatibility

Thanks

**Comment by cfenollosa** [comment]
Closing after 2 months

**Comment by Puuhinen** [comment]
I also have this issue on Linux (not sure of the distro on the server), awk is GNU Awk 4.0.1. I think the bug has to do with tags after a cut?

**Comment by cfenollosa** [comment]
Hi Puuhinen,

could you please provide an example usage that reproduces the bug? I can't trigger it with OP's comments

Thanks

**Comment by Enneka** [comment]
Hi, I have the same issue as aaronstonedd and Puuhinen.

Starting with a fresh install, with nano as my editor and markdown installed. I start with a `./bb.sh post` and write a basic entry.

If I check the generated .html file, it contains my tags. Editing with `./bb.sh edit my-title.md` cause no problem too.

But if I edit the html file with `./bb.sh edit my-title.html`, the html file will not contain any tag at the bottom. Same thing with `./bb.sh rebuild`, it will erase tags from html files.

As aaronstonedd found out, the problem is that in the generated html file, the tags are on the same line than `<!-- text end -->`. So when is it time to get the entry content in the html file, the tags line is cut (due to `while (getline > 0 && !/<!-- text end -->/) {`).

To fix the problem I added a newline : `echo '<!-- text end -->' >> "$filename"` becomes `echo -e '\n<!-- text end -->' >> "$filename"`.

I took the liberty to fork, 'fix' and make a pull request.
